### PR TITLE
Integral Image

### PR DESCRIPTION
**Pull request by syedTabish**
Integral Image PR on separate branch.
In response to #949

### CODE REVIEW

**Comment by syedTabish** [comment]
@stefanv, @jni travis is failing on this for a seemingly unrelated reason. I don't know how to correct this. 
Can this be merged??

**Comment by jni** [comment]
@syedTabish this Travis failure is repeatable and has appeared in other PRs... The answer to your question is "probably, but not until we figure out this failure and make it go away". =)

Thanks for this update, and thanks for your patience while we all argued over the best interface. =)

**Comment by emmanuelle** [comment]
what is the situation of this PR? Could we consider merging it?

**Comment by jni** [comment]
@emmanuelle Yep. =D

@syedTabish Did you think this day would come? =D Thank you so much for an excellent contribution! =)

File: skimage/transform/integral.py [hunk 1]
```diff
@@ -27,44 +28,117 @@ def integral_image(x):
...
 
+    Examples
+    --------
+    >>> arr = np.ones((5, 6), dtype=np.float)
+    >>> ii = integral_image(arr)
+    >>> integrate(ii, [(1, 0)], [(1, 2)])  # sum from (1,0) -> (1,2)
+    array([ 3.])
+    >>> integrate(ii, [(3, 3)], [(4, 5)])  # sum form (3,3) -> (4,5)
+    array([ 6.])
+    >>> integrate(ii, [(1, 0), (3, 3)], [(1, 2), (4, 5)])  # sum from (1,0) -> (1,2) and (3,3) -> (4,5)
```
Comment (ahojnnes) [inline skimage/transform/integral.py]: Wrap at 80 or separate code by comments.

File: skimage/transform/integral.py [hunk 2]
```diff
@@ -27,44 +28,117 @@ def integral_image(x):
...
+    array([ 3.,  6.])
+    """
+    rows = 1
+    # handle input from new input format
+    if len(args) == 0:
+        if isinstance(start, collections.Iterable):
+            rows = len(start)
+        start = np.array(start)
+        end = np.array(end)
+    # handle deprecated input format
```
Comment (ahojnnes) [inline skimage/transform/integral.py]: This deprecation should be documented and removed in next release.

**Comment by ahojnnes** [comment]
Two more comments, that should be fixed.

**Comment by syedTabish** [comment]
@jni  I can die peacefully now =)

**Pull request merged**